src/native/fs/gpfs.h

```diff
@@ -1215,10 +1219,35 @@ gpfs_stat(const char *pathname, /* File pathname */
  *              EBADF   Bad file desc
  *              EINVAL  Not a GPFS file
  *              ESTALE  cached fs information was invalid
+ *              EEXIST  Destination file already exists and not the same as
+ *                      in replace fd
  *-------------------------------------------------------------------------*/
 int GPFS_API
 gpfs_linkat(gpfs_file_t olddirfd, const char *oldpath,
-            gpfs_file_t newdirfd, const char *newpath, int flags);
+            gpfs_file_t newdirfd, const char *newpath, int flags,
+            gpfs_file_t replacefd);
```

Comment:
@eshelmarc @romayalon is there a versioning on this header? it would have to match the runtime library in order to work, right? should we make this dependency more explicit somewhere so that we will get a proper error (eg. libgpfs version mismatch...) vs just an error that a symbol was not found.

Reply:
The libgpfs code include a IFILE_MAGIC number that GPFS main code will verify.
Also GPFS development did not want to change en existing interface, namely gpfs_linkat() so we now have
**gpfs_linkat**(gpfs_file_t olddirfd, const char *oldpath, gpfs_file_t newdirfd, const char *newpath, int flags);
when you don't have a second fd, and
**gpfs_linkatif**(gpfs_file_t olddirfd, const char *oldpath, gpfs_file_t newdirfd, const char *newpath, int flags, gpfs_file_t replacefd);
when you do have a 2nd fd

Reply:
no